sbor/src/decode.rs

```diff
@@ -87,16 +87,32 @@ impl<'de> Decoder<'de> {
     }
 
     pub fn read_type(&mut self) -> Result<u8, DecodeError> {
-        self.read_u8()
+        self.read_byte()
     }
 
-    pub fn read_len(&mut self) -> Result<usize, DecodeError> {
+    pub fn read_variant_index(&mut self) -> Result<u8, DecodeError> {
+        self.read_byte()
+    }
+
+    pub fn read_variant_label(&mut self) -> Result<String, DecodeError> {
+        let n = self.read_dynamic_size()?;
+        let slice = self.read_bytes(n)?;
+        String::from_utf8(slice.to_vec()).map_err(|_| DecodeError::InvalidUtf8)
+    }
+
+    pub fn read_static_size(&mut self) -> Result<usize, DecodeError> {
```

Comment:
Just a thought - this probably shouldn't be a pub - because it should be protected behind a with_static_info check -- realistically, for now, people would just use `check_static_size` I imagine?

Reply:
This method allows developers to implement their custom codec. It's not part of the `Decode` trait.

Reply:
Ahh okay, well then I think it should include a check of whether the `with_static_info` boolean is active before reading? (and conversely for writing?)

Reply:
I'd kinda advocate for a separate encoder/decoder for with/without type information to remove these runtime checks, but that would be a fun refactor adding in generics everywhere :')